snmp/tests/test_unit.py

```diff
@@ -228,3 +248,14 @@ def test_cache_building(write_mock, read_mock):
         check._running = False
 
     write_mock.assert_called_once_with('', '["192.168.0.1"]')
+
+
+def test_trie():
+    trie = OIDTrie()
+    trie.set((1, 2), 'bar')
+    trie.set((1, 2, 3), 'foo')
+    assert trie.match((1,)) == ((1,), None)
+    assert trie.match((1, 2)) == ((1, 2), 'bar')
+    assert trie.match((1, 2, 3)) == ((1, 2, 3), 'foo')
+    assert trie.match((1, 2, 3, 4)) == ((1, 2, 3), 'foo')
```

Comment:
why does it return foo if no exact match?

Reply:
For table support, the name we pass is a name of a root (eg `ifErrors`), where leaves are the actual values, with indexes. Sometimes OIDs are also indexed `0` for weird reasons.